src/algebra/algebra/spectrum.lean

```diff
@@ -116,6 +124,34 @@ lemma resolvent_eq {a : A} {r : R} (h : r ∈ resolvent_set R a) :
   resolvent a r = ↑h.unit⁻¹ :=
 ring.inverse_unit h.unit
 
+lemma units_smul_resolvent {r : Rˣ} {s : R} {a : A} :
+  r • resolvent a (s : R) = resolvent (r⁻¹ • a) (r⁻¹ • s : R) :=
+begin
+  by_cases h : s ∈ spectrum R a,
+  { rw [mem_iff] at h,
+    simp only [resolvent, algebra.algebra_map_eq_smul_one] at *,
+    rw [smul_assoc, ←smul_sub],
+    have h' : ¬ is_unit (r⁻¹ • (s • 1 - a)),
+      from λ hu, h (by simpa only [smul_inv_smul] using is_unit.smul r hu),
+    simp only [ring.inverse_non_unit _ h, ring.inverse_non_unit _ h', smul_zero] },
+  { simp only [resolvent],
+    have h' : is_unit (r • (algebra_map R A (r⁻¹ • s)) - a),
+      { simpa [algebra.algebra_map_eq_smul_one, smul_assoc] using not_mem_iff.mp h },
+    rw [←h'.coe_sub_inv_smul, ←(not_mem_iff.mp h).unit_spec, ring.inverse_unit, ring.inverse_unit,
+      h'.coe_inv_sub_inv_smul],
+    simp only [algebra.algebra_map_eq_smul_one, smul_assoc, smul_inv_smul], },
+end
+
+lemma units_smul_resolvent_self {r : Rˣ} {a : A} :
+  r • resolvent a (r : R) = resolvent (r⁻¹ • a) (1 : R) :=
+by simpa only [units.smul_def, algebra.id.smul_eq_mul, units.inv_mul]
+  using @units_smul_resolvent _ _ _ _ _ r r a
+
+/-- The resolvent is a unit when the argument is in the resolvent set. -/
+lemma is_unit_resolvent {r : R} {a : A} (h : r ∈ resolvent_set R a) :
+  is_unit (resolvent a r) :=
+(resolvent_eq h).symm ▸ ⟨⟨(h.unit⁻¹ : Aˣ), h.unit, _, _⟩, rfl⟩
```

Comment:
Should we make this an iff:
```suggestion
lemma mem_resolvent_set_iff_is_unit {r : R} {a : A} :
  r ∈ resolvent_set R a ↔ is_unit (resolvent a r) :=
⟨λ h, (resolvent_eq h).symm ▸ ⟨⟨(h.unit⁻¹ : Aˣ), h.unit, _, _⟩, rfl⟩, λ h,
begin
  sorry,
end⟩
```

Reply:
yes, which caused me to realize `is_unit_ring_inverse` should be a thing, and it is, so the proof is trivial. I'm keeping `is_unit_resolvent` for ease of rewriting.